packages/web/src/pages/user/[uid].tsx

```diff
@@ -38,22 +41,47 @@ const UserProfilePage: NextPage = () => {
     fetchUser();
   }, [uid]);
 
+  React.useEffect(() => {
+    const checkUser = async () => {
+      try {
+        const res = await fetch('/api/users/me');
+
+        if (res.ok) {
```

Comment:
What happens if this isn't ok? Looks like we're just doing `setDidCheckCurrentUser(true)` regardless... is that intended?

EDIT: After seeing the use below, we probably should return from inside the `res.ok` block if we are using `didCheckCurrentUser` as an error